src/services/arweave/ArweaveCompositeDataProvider.ts

```diff
@@ -1,18 +1,50 @@
-import { ArNSContractState, SmartweaveContractSource } from '../../types';
+import { Warp, WarpFactory } from 'warp-contracts';
+
+import {
+  ANTContractState,
+  ArNSContractState,
+  ArweaveTransactionId,
+  SmartweaveContractSource,
+} from '../../types';
 
 export class ArweaveCompositeDataProvider implements SmartweaveContractSource {
   private _providers: SmartweaveContractSource[];
+  private _warp: Warp;
+  contractId: any;
   // TODO: implement strategy methods
   constructor(providers: SmartweaveContractSource[]) {
     this._providers = providers;
+    this._warp = WarpFactory.forMainnet();
```

Comment:
we shouldn't need warp here - the `WarpDataProvider` is passed in, and it has `WarpFactory` defined